src/dune_console/dune_console.mli

```diff
@@ -17,26 +17,77 @@ module Backend : sig
   val compose : t -> t -> t
 
   (** A dumb backend that hides the status line and simply dump the messages to
-      the terminal *)
+      the terminal. *)
   val dumb : t
 
-  (** A backend that just displays the status line in the terminal *)
-  val progress : unit -> t
-
+  (** A backend that displays the status line in the terminal. *)
+  val progress : t
+
+  (** A backend that displays the status line in the terminal, with the
+      processing logic happening in a separate thread. *)
+  val progress_threaded : unit -> t
+
+  (** Backends may have access to an internal state. This type is used by the
+      [threaded] backends. *)
+  type state =
+    { messages : User_message.t Queue.t
+    ; mutable finish_requested : bool
+    ; mutable finished : bool
+    ; mutable status_line : User_message.Style.t Pp.t option
+    }
+
+  (** [Threaded] is the interface for user interfaces that are rendered in a
+      separate thread. It can be used to direct construct a threaded backend. *)
```

Comment:
Should this be "It can be used to direct _and_ construct a threaded backend."?

Reply:
I meant to say directly. Will fix.